Upgrade to API v2

dfalmeida opened the pull request:
## Description
- Updated API to v2
- Updated packages dependencies to latest versions
- Updated README
- Do not run snapshot testing on CI to avoid comparison errors
- Changed the iOS test device from `iPhone 12 Pro Max` to `iPhone 13`

Commit:
chore(api): Update to API v2, Upgrade packages

Commit:
chore(api): Update Readme

Commit:
chore(api): Update Readme

Commit:
chore(fastlane): Updated project name

Commit:
chore: Updated packages

Commit:
fix(tests): New snapshot testing images

Commit:
fix(tests): changed test device to iPhone 13

dfalmeida committed:
chore(macOS-version): Try to use Mac OS 12

dfalmeida committed:
chore(macOS-version): rever to latest

dfalmeida committed:
chore(tests-output): Archive test results

dfalmeida committed:
chore(output-test): added test results output path

dfalmeida committed:
chore(tests-output): Archive test results

dfalmeida committed:
chore(tests-output): show simulator logs

dfalmeida committed:
chore(tests): SnapShot Tests output

Commit:
chore(tests): snap shot tests output folder xcode config

andreragsilva commented on PokemonCards/Application/Cards/CardsCore.swift:
```diff
@@ -115,21 +115,20 @@ let cardsReducer =
         )
 
       case .cardsResponse(.failure(let error)):
+        print(error)
```
Should we keep debug code? 🤔

dfalmeida replied:
Sure not! 😓

andreragsilva reviewed (changes requested):
@dfalmeida it also appears you changed the snapshot test screenshots by lowering the resolution. Was the high resolution affecting testing? This may affect the level of interface testing we have in terms of pixel perfect hence the curiosity 🤔

andreragsilva commented:
> @dfalmeida it also appears you changed the snapshot test screenshots by lowering the resolution. Was the high resolution affecting testing? This may affect the level of interface testing we have in terms of pixel perfect hence the curiosity 🤔

Also, tests are failing as you mentioned before, just leaving a heads up 👍

dfalmeida committed:
chore(debug): Debug actions

dfalmeida committed:
chore: Removed test-output folder

andreragsilva commented:
@dfalmeida CI is still failing 🤔  So I will wait till there is time to fix it so I can review once again 👍

dfalmeida committed:
chore(build): Changed to macos-11

Commit:
enhancement(snapshot): Changed snapshot strategy

Commit:
Merge branch 'enhancement/api-v2' of github.com:coletiv/coletiv-ios-swiftui-tca-example into enhancement/api-v2

andreragsilva commented:
@dfalmeida CI still failing so I will postpone my review once again, hope you are able to close this 💪 Let me know if you need help

dfalmeida committed:
chore(wf): revert test production

Commit:
chore(tests): Revert tests, do not run snapshot testing on CI

Commit:
Merge branch 'enhancement/api-v2' of github.com:coletiv/coletiv-ios-swiftui-tca-example into enhancement/api-v2

dfalmeida commented:
> @dfalmeida it also appears you changed the snapshot test screenshots by lowering the resolution. Was the high resolution affecting testing? This may affect the level of interface testing we have in terms of pixel perfect hence the curiosity 🤔

I changed the test device to a smaller one, maybe that was the cause 🤔

dfalmeida commented:
@andreragsilva just disabled the snapshot testing on CI. It is something that needs a deeper investigation. 
As far as I understood, the CI machines generates a pixelated snapshot that doesn't match the original one, I tried everything I could: 
* Changed to a different macOS version
* Changed the snapshot strategy
* Find a way to change the graphic requirements from the machine. Cannot find those settings ofc :/ 

In the end CI is green 🍏😑

andreragsilva reviewed (approved):
@dfalmeida lets close this, I believe when and if we pick this up again, a lot will have change so we will be able to find an easier path (hopefully) 👍

Great work anyway, hope you enjoyed it 💪
